### PR TITLE
[bitnami/redis] Revisit tests and remove extra pipelines

### PR DESCRIPTION
**Pull request by FraPazGal**
### Description of the change

After reviewing the existing tests and the general testing strategy we have been following until now, we saw there was a need to revisit them. The main objective that our chart testing strategy should aim for is asserting that the Chart deliverable works. As such, the focus should not be on the application itself or the container, which are regarded as trustful components (that is, they should have been tested at a previous stage).

As charts usually pack a number of different components, it is essential to test that their integrations and the application itself work as expected.

Consequently, we have updated the tests so they are related to the chart deliverable and left only a couple of basic tests to verify that the application functions correctly. Besides that, an effort has been made to reduce future tests' flakiness improving some existing selectors and checks.

Additionally, this revisit has reduced the number of executed pipelines for Redis to one, and thus its related CI file has been removed.

Tested in https://github.com/FraPazGal/charts/pull/121

### Checklist

- [x] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [x] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/master/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

File: .vib/redis/goss/goss.yaml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  {{ .Vars.master.persistence.path }}:
+    filetype: directory
+    exists: true
+    mode: "2775"
+    owner: root
+  /opt/bitnami/scripts/start-scripts:
+    filetype: directory
+    exists: true
+    mode: "2777"
+    owner: root
```
Comment (joancafom) [inline .vib/redis/goss/goss.yaml:45]: I would omit this path. There already are a bunch of other paths in the tests, but in particular, this one is used by the primary container to start the Redis service. An error related to this mounted volume would probably end up as a deployment failure.

File: .vib/redis/goss/goss.yaml [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  /opt/bitnami/redis/etc:
+    filetype: directory
+    exists: true
+    mode: "2777"
+    owner: root
+  /health:
+    filetype: directory
+    exists: true
+    mode: "2777"
+    owner: root
```
Comment (joancafom) [inline .vib/redis/goss/goss.yaml:60]: Same thoughts here.

File: .vib/redis/goss/goss.yaml [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+  {{- $key := printf "key_%s" (randAlpha 5) }}
+  {{- $value := printf "value_%s" (randAlpha 5) }}
+  redis-set-key-value-pairs:
+    exec: export {{ $auth }} && redis-cli {{ $master_endpoint }} SET {{ $key }} {{ $value }} {{ range $e, $i := until $replicas }} && redis-cli -h redis-replicas-{{ $i }}.redis-headless -p {{ $.Vars.replica.containerPorts.redis }} GET {{ $key }} | grep -q {{ $value }}{{ end }}
+    exit-status: 0
+  redis-disabled-command:
+    exec: {{ $auth }} redis-cli {{ $master_endpoint }} {{ .Vars.master.disableCommands }}
+    exit-status: 0
+    stdout:
+      - ERR unknown command
```
Comment (joancafom) [inline .vib/redis/goss/goss.yaml:15]: I would probably use a loop here, as `disableCommands` is originally an array.

**Review by joancafom — COMMENTED** [review]
Thanks for your PR @FraPazGal ! I left you some minor comments for you to review 😁

File: .vib/redis/goss/goss.yaml (hunk 3 above)
Comment (agomezmoron) [reply .vib/redis/goss/goss.yaml:15]: Good catch!

File: .vib/redis/goss/goss.yaml (hunk 1 above)
Comment (FraPazGal) [reply .vib/redis/goss/goss.yaml]: You are right, removed both of them.

**Review by agomezmoron — APPROVED** [review]

**Review by joancafom — APPROVED** [review]

**Pull request merged**